diff --git a/x-pack/plugin/security/src/test/java/org/elasticsearch/integration/FieldLevelSecurityTests.java b/x-pack/plugin/security/src/test/java/org/elasticsearch/integration/FieldLevelSecurityTests.java
index 849ec6987e2..0e39bcaa3b9 100644
--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/integration/FieldLevelSecurityTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/integration/FieldLevelSecurityTests.java
@@ -772,12 +772,12 @@ public class FieldLevelSecurityTests extends SecurityIntegTestCase {
     public void testScrollWithQueryCache() {
         assertAcked(client().admin().indices().prepareCreate("test")
                 .setSettings(Settings.builder().put(IndexModule.INDEX_QUERY_CACHE_EVERYTHING_SETTING.getKey(), true))
-                .setMapping("field1", "type=text", "field2", "type=text")
+                .addMapping("type1", "field1", "type=text", "field2", "type=text")
         );
 
         final int numDocs = scaledRandomIntBetween(2, 4);
         for (int i = 0; i < numDocs; i++) {
-            client().prepareIndex("test").setId(String.valueOf(i))
+            client().prepareIndex("test", "type1", String.valueOf(i))
                     .setSource("field1", "value1", "field2", "value2")
                     .get();
         }
