diff --git a/connector-packager/setup.py b/connector-packager/setup.py
index 1601492..eb1cc27 100644
--- a/connector-packager/setup.py
+++ b/connector-packager/setup.py
@@ -16,7 +16,7 @@ setup(
     description='A Python module for packaging a Tableau connector.',
     test_suite='tests',
     python_requires='>3.7',
-    install_requires=['xmlschema', 'defusedxml'],
+    install_requires=['elementpath==2.4.0', 'xmlschema==1.9.2', 'defusedxml==0.7.1'],
     tests_require=['six'],
     include_package_data=True
 )
